### DISCUSSION TITLE
`wasmedge_wasi_socket` crate seems to be broken?

### DISCUSSION THREAD

**Discussion by ilsubyeega** [WasmEdge/WasmEdge#1868] [Q&A]
```rs
use wasmedge_http_req::request;
fun main() {
  run_http_client();
}
fn run_http_client() -> io::Result<()> {
    let mut body = Vec::new();
    let res = request::get("http://18.207.88.57:80/get", &mut body) // httpbin ip addr
        .expect("Could not connect to that website.");

    println!("HTTP Status: {} {}", res.status_code(), res.reason());
    Ok(())
}
```

This one throws: `thread 'main' panicked at 'Could not connect to that website.: IO(Error { kind: WriteZero, message: "failed to write whole buffer" })'`, ive searched on issues and discussions on this repo, still didnt find the solution through.

cargo:
```
[dependencies]
wasmedge_wasi_socket = "0.4.1"
wasmedge_http_req  = "0.8.1"
```

Update:
Tested windows 11 native and ubuntu wsl, but both failed.

**Comment by alabulei1** [comment]
Hi @ilsubyeega 

Thanks for reporting this. Could you provide your WasmEdge version?

**Reply by ilsubyeega** [reply]
Hello, it seems that wasmedge version is 0.10.0
+) the latest version, 0.11.0 can be reproduced same

**Reply by alabulei1** [reply]
> Hello, it seems that wasmedge version is 0.10.0 +) the latest version, 0.11.0 can be reproduced same

Hi @ilsubyeega ,

Could you try the hyper HTTP client?  
https://github.com/WasmEdge/wasmedge_hyper_demo/tree/main/client

**Reply by ilsubyeega** [reply]
windows 11 native does not seem to work at this moment.
```shell
PS C:\Users\sob\Project\GitHub\wasmedge_hyper_demo\client> cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.28s
     Running `wasmedge target\wasm32-wasi\debug\wasmedge_hyper_client.wasm`
create client
thread 'main' panicked at 'failed to park: Os { code: 52, kind: Unsupported, message: "Function not implemented" }', C:\Users\sob\.cargo\git\checkouts\tokio-11011dab5e670df4\4f2e423\tokio\src\runtime\scheduler\current_thread.rs:308:31
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2022-09-24 19:02:36.297] [error] execution failed: unreachable, Code: 0x89
[2022-09-24 19:02:36.298] [error]     In instruction: unreachable (0x00) , Bytecode offset: 0x006f6007
[2022-09-24 19:02:36.299] [error]     When executing function name: "_start"
error: process didn't exit successfully: `wasmedge target\wasm32-wasi\debug\wasmedge_hyper_client.wasm` (exit code: 1)
```

**Comment by hydai** [comment]
Seems like `wasmedge_http_req` is updated to `wasmedge_wasi_socket 0.4.1`, however, the updated one is not published. Will check it later once I am back to the office.
